### PR TITLE
#60: Support Elasticsearch 8

### LINKED ISSUE #60: Adapt to Elasticsearch 8
Integration tests currently use Elasticsearch 7.17.5. Upgrading server and JDBC driver to the latest 8 release (currently 8.3.3), integration tests start failing because the JDBC driver returns slightly different values than before.

Until this is implemented we recommend using Elasticsearch version 7.17.5 for both the server and the JDBC driver.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #60

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialect.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ protected RemoteMetadataReader createRemoteMetadataReader() {
 
     @Override
     protected QueryRewriter createQueryRewriter() {
-        return new ImportFromJDBCQueryRewriter(this, createRemoteMetadataReader());
+        return new ImportIntoTemporaryTableQueryRewriter(this, createRemoteMetadataReader(), this.connectionFactory);
```
Comment (kaklakariada) [inline src/main/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialect.java:127]: This ensures that `IMPORT INTO` statements contain the expected data types.

File: src/test/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialectIT.java [hunk 2]
```diff
@@ -67,10 +71,11 @@ private static ElasticsearchContainer createElasticsearchContainer() {
         final ElasticsearchContainer container = new ElasticsearchContainer(ELASTICSEARCH_DOCKER_IMAGE_REFERENCE);
         // Disable TLS for Elasticsearch server as the self-signed certificate uses the wrong hostname and we can't
         // configure a HostnameVerifier for the JDBC driver.
-        container.withEnv("xpack.security.enabled", "false");
-        container.withEnv("discovery.type", "single-node");
+        container.withEnv("ES_SETTING_XPACK_SECURITY_ENABLED", "false");
+        container.withEnv("ES_SETTING_DISCOVERY_TYPE", "single-node");
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialectIT.java:75]: See https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docker.html#docker-configuration-methods how to configure ES in Docker via env variables

File: src/test/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialectIT.java [hunk 3]
```diff
@@ -264,6 +269,14 @@ void testDocumentWithNestedProperty() throws IOException {
                 table().row("str", "inner_str").matches(TypeMatchMode.NO_JAVA_TYPE_CHECK));
     }
 
+    @Test
+    void testSelectAllColumns() throws IOException {
+        this.indexDocument(createObjectBuilder().add("c1", "str").add("c2", 42).add("c3", 3.14).build());
+        final String query = "SELECT * FROM " + getVirtualTableName();
+        assertVirtualTableContentsByQuery(query,
+                table().row("str", "str", 42, 3.140000104904175).matches(TypeMatchMode.NO_JAVA_TYPE_CHECK));
+    }
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialectIT.java:278]: There was a note in the user guide saying that `SELECT *` did not work, so I added a test.

Comment (redcatbear) [reply src/test/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchSqlDialectIT.java:276]: :+1:

**Review by redcatbear — APPROVED** [review]

**Pull request merged**